src/Sylius/Behat/Context/Api/Shop/CartContext.php

```diff
@@ -530,7 +530,7 @@ public function itsPriceShouldBeDecreasedBy(ProductInterface $product, int $amou
     {
         $pricing = $this->getExpectedPriceOfProductTimesQuantity($product);
 
-        $this->compareItemPrice($product->getName(), $pricing - $amount);
+        $this->compareItemPrice($product->getName(), $pricing - $amount, 'discountedUnitPrice');
```

Comment:
We were checking `total` which is wrong and the return statement was called at the first iteration always ;d